Shellcheck emits warnings on git hook example in the documentation

<!--
Thank you for helping to improve terraform-docs!

Please be sure to search for open issues before raising a new one. We use issues
for bug reports and feature requests. Please note, this template is for bugs
report, not feature requests. For more information, see the Contributing Guidelines
at https://git.io/JtEzg. Please find us at https://slack.terraform-docs.io/ for
questions, support, and discussion.
-->

### Describe the bug

<!--
Please let us know what behaviour you expected and how terraform-docs diverged
from that behaviour.
-->

The [git hook example](https://terraform-docs.io/user-guide/how-to/#git-hook) in the [how-to docs](https://github.com/terraform-docs/terraform-docs/blob/master/docs/user-guide/how-to.md) has warnings when [shellcheck](https://github.com/koalaman/shellcheck) is ran against it.

### How can we reproduce it?

<!--
Help us to reproduce your bug as succinctly and precisely as possible. Any and
all steps or script that triggers the issue are highly appreciated!

Do you have long logs to share? Please go to https://ghostbin.com and insert the
link here.
-->

1. [install shellcheck](https://github.com/koalaman/shellcheck#installing)

2. save the hook code in a file, e.g. `./tmp.sh`:

```sh
#!/bin/sh

# Keep module docs up to date
for d in $(ls -1 modules)
do
  terraform-docs md modules/$d > modules/$d/README.md
  if [ $? -eq 0 ] ; then
    git add "./modules/$d/README.md"
  fi
done
```

3. run `shellcheck ./tmp.sh`

You will see the following output:

```txt
In tmp.sh line 4:
for d in $(ls -1 modules)
         ^--------------^ SC2045: Iterating over ls output is fragile. Use globs.


In tmp.sh line 6:
  terraform-docs md modules/$d > modules/$d/README.md
                            ^-- SC2086: Double quote to prevent globbing and word splitting.
                                         ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  terraform-docs md modules/"$d" > modules/"$d"/README.md


In tmp.sh line 7:
  if [ $? -eq 0 ] ; then
       ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

For more information:
  https://www.shellcheck.net/wiki/SC2045 -- Iterating over ls output is fragi...
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
  https://www.shellcheck.net/wiki/SC2181 -- Check exit code directly with e.g...
```

### Environment information

<!--
- terraform-docs version (use `terraform-docs --version`):
- Go version (if you manually built. use `go version`):
- OS (e.g. Windows, MacOS):
-->

```txt
terraform-docs version 0.11.2 bd351fc linux/amd64 BuildDate: 2021-02-22T23:52:49Z
```

### Proposed fix

The example code is:

```sh
#!/bin/sh

# Keep module docs up to date
for d in $(ls -1 modules)
do
  terraform-docs md modules/$d > modules/$d/README.md
  if [ $? -eq 0 ] ; then
    git add "./modules/$d/README.md"
  fi
done
```

Applying the shellcheck recommended changes, it becomes:

```sh
#!/bin/sh

# Keep module docs up to date
for d in modules/*; do
  if terraform-docs md "$d" > "$d/README.md"; then
    git add "./$d/README.md"
  fi
done
```

I can open a PR with this change to the docs if you'd like.

Thanks!

--- a/docs/user-guide/how-to.md
+++ b/docs/user-guide/how-to.md
@@ -111,11 +111,9 @@ make a commit. See also [git hooks] documentation.
 #!/bin/sh
 
 # Keep module docs up to date
-for d in $(ls -1 modules)
-do
-  terraform-docs md modules/$d > modules/$d/README.md
-  if [ $? -eq 0 ] ; then
-    git add "./modules/$d/README.md"
+for d in modules/*; do
+  if terraform-docs md "$d" > "$d/README.md"; then
+    git add "./$d/README.md"
   fi
 done
 ```

